Fixed bug when ES is down.

rajgiridhar1995 opened the pull request.

Commit:
Add elasticsearch 6.2.4 support

rajgiridhar1995 committed:
# This is a combination of 7 commits.
# This is the 1st commit message:

Fixed bug when ES is down.

# This is the commit message #2:

ES version updated to 6.5.2

# This is the commit message #3:

Removed .DS_Store and other relevant files.

# This is the commit message #4:

Removed .DS_Store and other relevant files.

# This is the commit message #5:

Removed .DS_Store and other relevant files.

# This is the commit message #6:

Removed .DS_Store and other relevant files.

# This is the commit message #7:

Removed unnecessary class files.

ankitnanglia commented on pom.xml:
```diff
@@ -43,6 +43,14 @@
+            </plugin>
         </plugins>
     </build>
     <dependencies>
```
I see a lot .DS_Store file being committed as part of this change. 
Can you remove them and also update the .gitignore file to prevent this file from being added to the git

rajgiridhar1995 replied:
removed .DS_Store

ankitnanglia commented on src/test/java/com/cognitree/flume/sink/elasticsearch/TestAvroSerializer.java:
```diff
@@ -100,7 +101,7 @@ public void testSerializer() throws Exception {
         XContentBuilder actual = avroSerializer.serialize(event);
         JsonParser parser = new JsonParser();
-        assertEquals(parser.parse(expected.string()), parser.parse(actual.string()));
+        assertEquals(parser.parse(Strings.toString(expected)), parser.parse(Strings.toString(actual)));
```
Was this working before? Why did we change this now?

rajgiridhar1995 replied:
Reverted to original version

ankitnanglia commented on src/main/java/com/cognitree/flume/sink/elasticsearch/client/BulkProcessorBuilder.java:
```diff
@@ -112,8 +115,23 @@ public void afterBulk(long executionId,
+    private void checkNodeConnectionTimer(){
+        while(true) {
+            try {
+                wait(3000);
```
extract as static constant

ankitnanglia commented on pom.xml:
```diff
@@ -25,7 +25,7 @@
 
     <properties>
-        <elasticsearch-verison>6.2.4</elasticsearch-verison>
+        <elasticsearch-verison>6.5.2</elasticsearch-verison>
```
Why did we change the version? I thought this Merge Request is for the bug fix?

rajgiridhar1995 replied:
Reverted to original version.

ankitnanglia commented on pom.xml:
```diff
@@ -43,6 +43,14 @@
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>6</source>
```
Why is the source and target set to 6? Do you what 6 means here

rajgiridhar1995 replied:
REMOVED.

rajgiridhar1995 committed:
Fixed bug when ES is down.

ES version updated to 6.5.2

Removed .DS_Store and other relevant files.

Removed .DS_Store and other relevant files.

Removed .DS_Store and other relevant files.

Removed .DS_Store and other relevant files.

Removed unnecessary class files.

Removed unnecessary files.

Removed unnecessary class files.

Removed unnecessary files.

Added method description and rename of variables.

ankitnanglia commented on src/main/java/com/cognitree/flume/sink/elasticsearch/ElasticSearchSink.java:
```diff
@@ -33,6 +33,10 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.Timer;
```
Can you squash the number of commits?
I see a lot of commits with same commit message. 
Ideally should be a single commit for the bug fix.
Look for squashing multiple commits into one.

rajgiridhar1995 committed:
Fixed the bug by throwing an exception instead of stopping the
consumption from the channel.

rajgiridhar1995 committed:
Merge branch 'master' of github.com:rajgiridhar1995/flume-elasticsearch-sink into HEAD
